ci/run.sh

```diff
@@ -100,12 +101,20 @@ test() {
 
 # test dmd
 test_dmd() {
-    # test fewer compiler argument permutations for PRs to reduce CI load
-    if [ "$FULL_BUILD" == "true" ] && [ "$OS_NAME" == "linux"  ]; then
-        make -j1 -C compiler/test MODEL=$MODEL N=$N # all ARGS by default
+    # run the d_do_test unittests
+    $build_path/dmd -m$MODEL -g -i -Icompiler/test/tools -version=NoMain -lowmem -unittest -run compiler/test/tools/d_do_test.d
```

Comment:
The Makefile is deprecated for years now AFAIK, but these `d_do_test` unittests are still exclusively in the Makefile, with an according TODO: https://github.com/dlang/dmd/blob/428cc6d3c4064cff21e3b823119b03ab380279bd/compiler/test/Makefile#L226-L231

Reply:
Tackled in https://github.com/dlang/dmd/pull/15834.